src/RpcServer/RpcServer.Wallet.cs

```diff
@@ -348,20 +347,20 @@ private JObject GetVerificationResult(UInt160 scriptHash, ContractParameter[] ar
             {
                 throw new RpcException(-100, "Unknown contract");
             }
-            var methodName = "verify";
+            var md = contract.Manifest.Abi.GetMethod("verify", 0);
+            if (md is null)
+                throw new RpcException(-101, $"The smart contract {contract.Hash} haven't got verify method without arguments");
+            if (md.ReturnType != ContractParameterType.Boolean)
+                throw new RpcException(-102, "The verify method doesn't return boolean value.");
 
             Transaction tx = signers == null ? null : new Transaction
             {
                 Signers = signers.GetSigners(),
-                Attributes = Array.Empty<TransactionAttribute>()
+                Attributes = Array.Empty<TransactionAttribute>(),
+                Witnesses = signers.Witnesses,
```

Comment:
`Witnesses` it wil lbe null, isn't it?

Reply:
Yes, just like https://github.com/neo-project/neo-modules/blob/93fb62019bec6606c406e337706d5c5e235f948c/src/RpcServer/RpcServer.SmartContract.cs#L69

Reply:
Removed it since no use here.

Reply:
Can we pass witnesses via RPC request parameters, just like signers are passed? And then fill `Witnesses` here.